src/utils/validation.ts

```diff
@@ -27,23 +27,23 @@ interface FormDataAsNumbers {
  * @param validationSchema joi.ObjectSchema<unknown> - Joi schema to check data
  * @param type [OPTIONAL] 'number' | undefined - sets casting use or straight FormData use
  */
-export const stringOrNumberResolverFactory = <FormData, CustomContext = {}>(
-  validationSchema: ObjectSchema<unknown>,
-  type?: 'number',
-) => (data: FormData): ReturnType<ValidationResolver<FormData, CustomContext>> => {
-  const castedData: FormDataAsNumbers | FormData =
-    type === 'number'
-      ? Object.keys(data).reduce<FormDataAsNumbers>((acc, key) => {
-          const oldValue = data[key]
-          const castedValue = Number(oldValue)
-          const castedObj = { ...acc, [key]: castedValue }
 
-          return castedObj
-        }, {})
-      : data
+export const resolverFactory = <FormData, CustomContext = {}>(validationSchema: ObjectSchema<unknown>) => (
+  data: FormData,
+): ReturnType<ValidationResolver<FormData, CustomContext>> => {
+  const castedData: FormDataAsNumbers | FormData = Object.keys(data).reduce<FormDataAsNumbers>((acc, key) => {
+    const oldValue = data[key]
+    const castedValue = oldValue ? oldValue : undefined
+    const castedObj = { ...acc, [key]: castedValue }
+
+    return castedObj
+  }, {})
```

Comment:
What happens here?
```
data = {a: '', b: 0, c: false, e: Truthy}
castedData = {a: undefined, b: undefined, c: undefined, : Truthy}
```
Is that what you want? How does validation treat it? Skips undefined keys?

Reply:
falsy = explicit undefined, was some issue with `FormData` taking undefined values and setting as `""` which `schemaValidation` wasn't treating as `undefined`